### PR TITLE
fix(emails): Allow users to login with a secondary email

### PR DESCRIPTION
**Pull request by vbudhram**
Lets see what breaks 🤔

Connects #3427

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Oh this is amazing if it works!

File: packages/fxa-auth-server/test/remote/recovery_email_emails.js [hunk 1]
```diff
@@ -851,14 +851,28 @@ describe('remote emails', function() {
...
-    it('fails to login', () => {
+    it('successful login', () => {
       return Client.login(config.publicUrl, secondEmail, password, {})
         .then(() => {
           assert.fail(new Error('should not have been able to login'));
         })
         .catch(err => {
+          // When logging in with a secondary email the auth-server returns the correct email
+          // to generate the authPW (based on OnePW protocol). This is the email that should be
+          // used to login.
```
Comment (rfk) [inline packages/fxa-auth-server/test/remote/recovery_email_emails.js]: IIRC, one reason we haven't done this in the past is that it's a small unauthenticated info leak - if I know your secondary email address, I could do a fake login attempt and have the FxA server tell me your primary address. Have we / do we need to clear this info leak with security/privacy review?

**Comment by dannycoates** [comment]
This is a pretty significant change in behavior. I think we should be careful here and consider other options.

**Comment by vbudhram** [comment]
After speaking with the team and security folks, we are going not going to change this. Currently, not a lot of users experience this issue and therefore it might not have a big impact.

Ref: https://sentry.prod.mozaws.net/operations/fxa-content-product/issues/627683/?query=is:unresolved

**Pull request closed without merging**